gov2/demotools/question.go

```diff
@@ -131,12 +132,20 @@ func (questioner Questioner) Ask(question string, validators ...IAnswerValidator
 }
 
 // AskBool asks a question with an expected answer. If the expected answer is given,
-// it returns true; otherwise, it returns false.
+// it returns true; otherwise, it returns false unless the answer is empty, in which
+// case it will ask again.
 func (questioner Questioner) AskBool(question string, expected string) bool {
 	answer := questioner.Ask(question, NotEmpty{})
 	return strings.ToLower(answer) == expected
 }
 
+// AskBoolWithDefault asks a question with an expected answer. If the expected answer is given,
+// it returns true; otherwise, it returns false.
+func (questioner Questioner) AskBoolWithDefault(question string, expected string, defaultValue string) bool {
+	answer := questioner.Ask(question)
+	return strings.ToLower(answer) == expected || strings.ToLower(expected) == strings.ToLower(defaultValue)
+}
```

Comment:
This works fine, but I almost would prefer `defaultResponse bool` instead of `defaultValue` and then
```
return strings.ToLower(answer) == expected || defaultResponse
```

Reply:
This made me realize I was missing some test coverage for when the default response is affirmative but the provided response by the user is negative. I'll rewrite and then take a look at using a bool instead of a string.